CLAUDE.md

```diff
@@ -0,0 +1,142 @@
+# PyTorch Development Guide
+
+This is the main PyTorch repository - a Python package providing tensor computation with GPU acceleration and deep neural networks built on a tape-based autograd system.
+
+## 🏗️ Project Overview
+
+PyTorch is organized into several key components:
+- **`torch/`** - Main Python API, frontend and C++/Python binding code
+- **`aten/`** - ATen tensor library (C++ backend)
+- **`c10/`** - Core C++ utilities, types, and device abstraction
+- **`torchgen/`** - Code generation tools for operators used during build time
+- **`test/`** - Comprehensive test suite
+- **`tools/`** - Build tools and development utilities
+- **`functorch/`** - Functional transforms (vmap, grad, etc.), this was moved into torch/
+- **`benchmarks/`** - Performance benchmarking tools
+
+## 🚀 Quick Development Setup
+
+### Build PyTorch from Source
+```bash
+# Default fast CPU build configuration
+BUILD_CONFIG="CFLAGS='-DPYBIND11_DETAILED_ERROR_MESSAGES -DHAS_TORCH_SHOW_DISPATCH_TRACE' USE_DISTRIBUTED=0 USE_FLASH_ATTENTION=0 USE_MEM_EFF_ATTENTION=0 USE_MKLDNN=0 USE_CUDA=0 BUILD_TEST=0 USE_FBGEMM=0 USE_NNPACK=0 USE_QNNPACK=0 USE_XNNPACK=0 USE_COLORIZE_OUTPUT=0"
+
+# Install dependencies
+pip install -r requirements.txt
+
+# Build and install in development mode (editable install)
+eval $BUILD_CONFIG python setup.py develop
+
+# Clean build artifacts
+python setup.py clean
+```
+
+### Other Environment Variables for Building
+```bash
+# Debug build with symbols, very expensive, should almost never be used
+export DEBUG=1
+
+# Release with debug info
+export REL_WITH_DEB_INFO=1
+
+# Enable specific features
+export USE_CUDA=1         # Enable CUDA support
+export USE_DISTRIBUTED=1  # Enable distributed training
+export USE_CUSTOM_DEBINFO='/path/to/file;path/to/other/file' # Build debug symbold only for these files
```

Comment:
Yeah, like, toggling this features is a MAJOR thing in terms of build time and I generally don't trust the agent to decide if it should do this or not.